ScanCommand/src/test/java/io/ballerina/scan/TestScanCmd.java

```diff
@@ -0,0 +1,93 @@
+/*
+ *  Copyright (c) 2024, WSO2 LLC. (https://www.wso2.com).
+ *
+ *  WSO2 LLC. licenses this file to you under the Apache License,
+ *  Version 2.0 (the "License"); you may not use this file except
+ *  in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing,
+ *  software distributed under the License is distributed on an
+ *  "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ *  KIND, either express or implied. See the License for the
+ *  specific language governing permissions and limitations
+ *  under the License.
+ */
+
+package io.ballerina.scan;
+
+import org.apache.commons.lang3.SystemUtils;
+import org.testng.Assert;
+import org.testng.annotations.Test;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.nio.charset.StandardCharsets;
+import java.util.ArrayList;
+import java.util.List;
+
+public class TestScanCmd {
+    @Test
+    void testScanCommand() {
+        Process scanProcess = startProcess();
+        StringBuilder output = readProcessOutput(scanProcess);
+
+        int exitCode;
+        try {
+            exitCode = scanProcess.waitFor();
+        } catch (InterruptedException e) {
+            throw new RuntimeException(e);
+        }
+
+        if (exitCode == 0) {
+            // Retrieve and assert the first line
+            String firstLine = output.toString().trim().split("\n")[0];
+            Assert.assertEquals("Running Scans...", firstLine);
+        }
+    }
+
+    private Process startProcess() {
```

Comment:
Move to a separate test utils class. And rename to something like scan accepting relevant arguments for variable things. Basically methods you can use with future tests also.

Reply:
Scan tool tests have been refactored to work similar to tests of other Ballerina commands in [691499](https://github.com/ballerina-platform/static-code-analysis-tool/pull/3/commits/691499940e4f1889ce3b1b15e0c5abe412a4c2c1)